engine/src/state.rs

```diff
@@ -1,60 +1,160 @@
-use crate::parameters::NewCallArgs;
+use crate::parameters::{LegacyNewCallArgs, NewCallArgs, NewCallArgsV2};
 use aurora_engine_sdk::io::{StorageIntermediate, IO};
 use aurora_engine_types::account_id::AccountId;
 use aurora_engine_types::storage::{bytes_to_key, KeyPrefix};
+use aurora_engine_types::{Cow, Vec};
 use borsh::{BorshDeserialize, BorshSerialize};
 
 pub use error::EngineStateError;
 
 /// Key for storing the state of the engine.
-const STATE_KEY: &[u8; 5] = b"STATE";
+pub const STATE_KEY: &[u8; 5] = b"STATE";
 
 /// Engine internal state, mostly configuration.
 /// Should not contain anything large or enumerable.
-#[derive(BorshSerialize, BorshDeserialize, Default, Clone, PartialEq, Eq, Debug)]
+#[derive(Default, Clone, PartialEq, Eq, Debug)]
 pub struct EngineState {
     /// Chain id, according to the EIP-155 / ethereum-lists spec.
     pub chain_id: [u8; 32],
     /// Account which can upgrade this contract.
     /// Use empty to disable updatability.
     pub owner_id: AccountId,
-    /// Account of the bridge prover.
-    /// Use empty to not use base token as bridged asset.
-    pub bridge_prover_id: AccountId,
     /// How many blocks after staging upgrade can deploy it.
     pub upgrade_delay_blocks: u64,
 }
 
-impl From<NewCallArgs> for EngineState {
-    fn from(args: NewCallArgs) -> Self {
+impl EngineState {
+    pub fn to_bytes(&self) -> Result<Vec<u8>, error::EngineStateError> {
```

Comment:
The `to_bytes` is an ambiguous name, IMO.  I would rename it to `to_borsh_bytes` or at least add some comment docs to explain what bytes we get here.

Reply:
Fixed in https://github.com/aurora-is-near/aurora-engine/pull/749/commits/b1015b0fe9e51a30ecd5ae4b2034faaf6fbd26f3